src/bio/bio_buffer.c

```diff
@@ -1046,6 +1052,13 @@ nvme_rw(struct bio_desc *biod, struct bio_rsrvd_region *rg)
 	if (daos_io_bypass & IOBP_NVME)
 		return;
 
+	/* No locking for BS state query here is tolerable */
+	if (xs_ctxt->bxc_blobstore->bb_state == BIO_BS_STATE_FAULTY) {
+		D_ERROR("Blobstore is marked as FAULTY.\n");
+		biod->bd_result = -DER_NVME_IO;
```

Comment:
Check glb_criteria.fc_enabled for update case before set NVME_IO?

Reply:
Ok.